src/components/ParentNavigationSubtitle.tsx

```diff
@@ -43,6 +61,31 @@ function ParentNavigationSubtitle({parentNavigationSubtitleData, parentReportAct
             onPress={() => {
                 const parentAction = getReportAction(parentReportID, parentReportActionID);
                 const isVisibleAction = shouldReportActionBeVisible(parentAction, parentAction?.reportActionID ?? CONST.DEFAULT_NUMBER_ID, canUserPerformWriteAction);
+
+                if (openParentReportInCurrentTab && isReportInRHP) {
+                    // If the report is displayed in RHP in Reports tab, we want to stay in the current tab after opening the parent report
+                    if (currentFullScreenRoute?.name === NAVIGATORS.SEARCH_FULLSCREEN_NAVIGATOR) {
+                        const lastRoute = currentFullScreenRoute?.state?.routes.at(-1);
+                        if (lastRoute?.name === SCREENS.SEARCH.MONEY_REQUEST_REPORT) {
+                            const moneyRequestReportID = (lastRoute?.params as SearchFullscreenNavigatorParamList[typeof SCREENS.SEARCH.MONEY_REQUEST_REPORT])?.reportID;
+                            // If the parent report is already displayed under RHP, simply dismiss the modal
```

Comment:
Maybe this sounds a bit more specific? I dunno honestly, took me like a minute to understand what "under" means in this context.
```suggestion
                            // If the parent report is already displayed underneath RHP, simply dismiss the modal
```

or maybe `below` could also work